sample/benchmark/src/test/java/com/datadog/benchmark/sample/config/DatadogFeaturesInitializerTest.kt

```diff
@@ -0,0 +1,147 @@
+        val config =
+            BenchmarkConfig(run = SyntheticsRun.Instrumented, scenario = SyntheticsScenario.SessionReplay)
+
+        mockAllAndCheck(config) {
+            // Then
+            sessionReplay.verifySessionReplayEnabled()
+            rum.verifyRumEnabled()
+            logs.verifyNoInteractions()
+        }
+    }
+
+    @Test
+    fun `M skip session replay and enable rum W run is baseline and scenario is sr`() {
+        // Given
+        val config = BenchmarkConfig(
+            run = SyntheticsRun.Baseline,
+            scenario = SyntheticsScenario.SessionReplay
+        )
+
+        mockAllAndCheck(config) {
+            // Then
+            sessionReplay.verifyNoInteractions()
+            rum.verifyRumEnabled()
+            logs.verifyNoInteractions()
+        }
+    }
+
+    @Test
+    fun `M enable nothing W scenario is logsCustom and run is baseline`() {
+        // Given
+        val config = BenchmarkConfig(
+            run = SyntheticsRun.Baseline,
+            scenario = SyntheticsScenario.LogsCustom
+        )
+
+        mockAllAndCheck(config) {
+            // Then
+            sessionReplay.verifyNoInteractions()
+            rum.verifyNoInteractions()
+            logs.verifyNoInteractions()
+        }
+    }
+
+    @Test
+    fun `M enable logs W scenario is logsCustom and run is instrumented`() {
+        // Given
+        val config = BenchmarkConfig(
+            run = SyntheticsRun.Instrumented,
+            scenario = SyntheticsScenario.LogsCustom
+        )
+
+        mockAllAndCheck(config) {
+            // Then
+            sessionReplay.verifyNoInteractions()
+            rum.verifyNoInteractions()
+            logs.verifyLogsEnabled()
+        }
+    }
+
+    private fun mockAllAndCheck(config: BenchmarkConfig, thenBlock: MockedStatics.() -> Unit) {
+        Mockito.mockStatic(SessionReplay::class.java).use { sessionReplayStatic ->
+            Mockito.mockStatic(Rum::class.java).use { rumStatic ->
+                Mockito.mockStatic(Logs::class.java).use { logsStatic ->
+                    // When
+                    DatadogFeaturesInitializer(
+                        sdkCore = mockSdkCore,
+                        config = config
+                    ).initialize()
+
+                    // Then
+                    thenBlock(MockedStatics(sessionReplayStatic, rumStatic, logsStatic))
+                }
+            }
+        }
+    }
+}
+
+private class MockedStatics(
+    val sessionReplay: MockedStatic<SessionReplay>,
+    val rum: MockedStatic<Rum>,
+    val logs: MockedStatic<Logs>
+)
+
+private fun MockedStatic<SessionReplay>.verifySessionReplayEnabled() {
+    verify {
+        SessionReplay.enable(
+            sessionReplayConfiguration = any(),
+            sdkCore = any()
+        )
+    }
+}
+
+private fun MockedStatic<Rum>.verifyRumEnabled() {
+    verify {
+        Rum.enable(
+            rumConfiguration = any(),
+            sdkCore = any()
+        )
+    }
+}
```

Comment:
The alternative to these checks is `Logs.isEnabled`, etc. and this will also allow to avoid using `mockStatic`, but then it won't be really a unit-test then.